### PR TITLE
Check that the correct number of bound values are provided

### PR DESCRIPTION
**Pull request by olim7t**

### CODE REVIEW

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java [hunk 1]
```diff
@@ -294,6 +294,9 @@ public Authenticator getAuthenticator() {
             ParsedStatement.Prepared prepared = QueryProcessor.parseStatement(cql, internalState);
             internalOptions.prepare(prepared.boundNames);
             CQLStatement statement = prepared.statement;
+            if (statement.getBoundTerms() != options.getValues().size()) {
+              throw new InvalidRequestException("Invalid amount of bind variables");
+            }
```
Comment (olim7t) [inline persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:299]: Copied from the respective `QueryProcessor` implementations.

**Comment by dougwettlaufer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java  1
- persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6276174&bid=20428668)

**Comment by mpenick** [comment]
I fixed this as part of https://github.com/stargate/stargate/pull/190

**Comment by olim7t** [comment]
Side note: no change needed for DSE, it worked already.

**Comment by olim7t** [comment]
@mpenick ah, no worries. I thought bound statement unset values would follow a different path, I should have checked your PR first. At least it looks like our fixes are similar :)

**Pull request closed without merging**